### PR TITLE
[Keyboard] Add Kudox Full

### PR DESCRIPTION
**Pull request by x1-**
## Description

Add `Kudox Full` as a keyboard that I designed recently.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* nothing

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kudox_full/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kudox_full/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kudox_full/keymaps/jis/keymap.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kudox_full/keymaps/jis/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kudox_full/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,109 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kudox_full/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kudox_full/readme.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# The Kudox Full keyboard
+
+<p align="center">
```
Comment (drashna) [inline keyboards/kudox_full/readme.md:3]: please remove all html code from the readme and use markdown, instead.

File: keyboards/kudox_full/rev1/config.h [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xABBA
+#define PRODUCT_ID      0x9693
+#define DEVICE_VER      0x0100
+#define MANUFACTURER    Kumao Kobo
+#define PRODUCT         The Kudox Full Keyboard
+#define DESCRIPTION     Split row staggered custom keyboard
```
Comment (drashna) [inline keyboards/kudox_full/rev1/config.h:27]:
```suggestion
```

File: keyboards/kudox_full/rev1/config.h [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN D3
+
+#undef RGBLED_NUM
+#define RGBLED_NUM 14    // Number of LEDs
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/kudox_full/rev1/config.h:64]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/kudox_full/rev1/info.json [hunk 7]
```diff
@@ -0,0 +1,85 @@
+{
+    "url": "http://kumaokobo.com/",
+    "maintainer": "Kumao Kobo",
+    "keyboard_name": "Kudox Full Keyboard Rev1",
+    "width": 19,
+    "height": 15,
```
Comment (drashna) [inline keyboards/kudox_full/rev1/info.json:6]:
```suggestion
```

File: keyboards/kudox_full/rev1/rev1.h [hunk 8]
```diff
@@ -0,0 +1,23 @@
+#pragma once
```
Comment (drashna) [inline keyboards/kudox_full/rev1/rev1.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kudox_full/rules.mk [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/kudox_full/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/kudox_full/rules.mk [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
```
Comment (drashna) [inline keyboards/kudox_full/rules.mk:23]:
```suggestion
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality 
RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
UNICODE_ENABLE = yes        # Unicode
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by x1-** [comment]
@drashna 
Thank you for reviewing my pull request!
I modified the codes you suggested.
Would you like to review again?

Best regards,

File: keyboards/kudox_full/rules.mk [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/kudox_full/rules.mk:21]:
```suggestion
```

File: keyboards/kudox_full/rev1/rev1.h [hunk 12]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 Kumao Kobo (@kumaokobo)
```
Comment (fauxpark) [inline keyboards/kudox_full/rev1/rev1.h:1]: rev1.c needs to include this file.

File: keyboards/kudox_full/readme.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+```sh
+make kudox_full/rev1:default
+```
+
+
+Example of flashing this keyboard:
+
+```sh
+make kudox_full/rev1:default:flash
+```
```
Comment (fauxpark) [inline keyboards/kudox_full/readme.md:24]:
```suggestion
    make kudox_full/rev1:default

Flashing example for this keyboard:

    make kudox_full/rev1:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by x1-** [comment]
@fauxpark 
Thank you for reviewing!
I modified you suggested.
Could you review it again?

Best regards,

File: keyboards/kudox_full/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+            // Or use the write_ln shortcut over adding '\n' to the end of your string
+            oled_write_ln_P(PSTR("\nLayer: Undefined"), false);
+    }
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+#endif
```
Comment (drashna) [inline keyboards/kudox_full/keymaps/default/keymap.c:122]:
```suggestion
```

File: keyboards/kudox_full/keymaps/jis/keymap.c [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+            // Or use the write_ln shortcut over adding '\n' to the end of your string
+            oled_write_ln_P(PSTR("\nLayer: Undefined"), false);
+    }
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+#endif
```
Comment (drashna) [inline keyboards/kudox_full/keymaps/jis/keymap.c:120]:
```suggestion
```

File: keyboards/kudox_full/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+            // Or use the write_ln shortcut over adding '\n' to the end of your string
+            oled_write_ln_P(PSTR("\nLayer: Undefined"), false);
+    }
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+#endif
```
Comment (drashna) [inline keyboards/kudox_full/keymaps/via/keymap.c:112]:
```suggestion
```

File: keyboards/kudox_full/kudox_full.c [hunk 17]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2021 Kumao Kobo (@kumaokobo)
+// SPDX-License-Identifier: GPL-2.0+
+
+#include "kudox_full.h"
```
Comment (drashna) [inline keyboards/kudox_full/kudox_full.c:4]:
```suggestion
#include "kudox_full.h"

#ifdef OLED_ENABLE
bool oled_task_kb(void) {
    if (!oled_task_user()) { return false; }
    // Host Keyboard Layer Status
    oled_write_ln_P(PSTR("Kudox Keyboard"), false);
    switch (get_highest_layer(layer_state)) {
        case 0:
            oled_write_ln_P(PSTR("\n"), false);
            break;
        case 1:
            oled_write_ln_P(PSTR("\nLayer: Symbol"), false);
            break;
        case 2:
            oled_write_ln_P(PSTR("\nLayer: Function"), false);
            break;
        case 3:
            oled_write_ln_P(PSTR("\nLayer: Light"), false);
            break;
        default:
            // Or use the write_ln shortcut over adding '\n' to the end of your string
            oled_write_ln_P(PSTR("\nLayer: Undefined"), false);
    }
    // Host Keyboard LED Status
    led_t led_state = host_keyboard_led_state();
    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);

    return true;
}
#endif
```

File: keyboards/kudox_full/rev1/config.h [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+#define MOUSEKEY_INTERVAL       20
+#define MOUSEKEY_DELAY          0
+#define MOUSEKEY_TIME_TO_MAX    60
+#define MOUSEKEY_MAX_SPEED      7
+#define MOUSEKEY_WHEEL_DELAY    0
```
Comment (drashna) [inline keyboards/kudox_full/rev1/config.h:83]: This would be better in `post_config.h`, with `ifndef`'s.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kudox_full/kudox_full.c (hunk 17 above)
Comment (x1-) [reply keyboards/kudox_full/kudox_full.c:4]: Thank you. That makes sense!

**Comment by x1-** [comment]
@drashna 
Thank you for reviewing!
Please review again 🙇

File: keyboards/kudox_full/rules.mk [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/kudox_full/rules.mk:16]:
```suggestion
```

File: keyboards/kudox_full/readme.md [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+
+![Kudox logo](https://raw.githubusercontent.com/kumaokobo/kudox-keyboard/master/img/kudox.png)
+
+![Kudox Full image2](https://raw.githubusercontent.com/kumaokobo/kudox-keyboard/master/img/kudox-full-image2.jpg) | ![Kudox Full image1](https://raw.githubusercontent.com/kumaokobo/kudox-keyboard/master/img/kudox-full-image1.jpg)
+:----:|:-----:
+ front | angle
+
+- Keyboard Maintainer: [Kumao Kobo](https://github.com/kumaokobo)  
+- Hardware Supported: Kudox Full PCB rev1.0 w/ Pro Micro  
+- Assembly guide: [Kudox Full/README](https://github.com/kumaokobo/kudox-keyboard/blob/master/kudox-full/README.md)
```
Comment (fauxpark) [inline keyboards/kudox_full/readme.md:11]:
```suggestion
- Keyboard Maintainer: [Kumao Kobo](https://github.com/kumaokobo)
- Hardware Supported: Kudox Full PCB rev1.0 w/ Pro Micro
- Assembly guide: [Kudox Full/README](https://github.com/kumaokobo/kudox-keyboard/blob/master/kudox-full/README.md)
```

File: keyboards/kudox_full/readme.md [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kudox_full/rev1:default
+
+
+Example of flashing this keyboard:
```
Comment (fauxpark) [inline keyboards/kudox_full/readme.md:18]:
```suggestion
Flashing example for this keyboard:
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by x1-** [comment]
@fauxpark 
Thank you for reviewing!
Please review again 🙇

File: keyboards/kudox_full/kudox_full.c [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+
+#ifdef RGBLIGHT_ENABLE
+void eeconfig_init_user(void) {
```
Comment (drashna) [inline keyboards/kudox_full/kudox_full.c:7]:
```suggestion
void eeconfig_init_kb(void) {
    eeconfig_update_kb(0);
```

File: keyboards/kudox_full/post_config.h [hunk 23]
```diff
@@ -0,0 +1,7 @@
+#ifndef MOUSEKEY_ENABLE
+    #define MOUSEKEY_INTERVAL       20
+    #define MOUSEKEY_DELAY          0
+    #define MOUSEKEY_TIME_TO_MAX    60
+    #define MOUSEKEY_MAX_SPEED      7
+    #define MOUSEKEY_WHEEL_DELAY    0
+#endif
```
Comment (drashna) [inline keyboards/kudox_full/post_config.h:7]:
```suggestion
// Copyright 2021 Kumao Kobo (@kumaokobo)
// SPDX-License-Identifier: GPL-2.0+

#pragma once

#ifndef MOUSEKEY_INTERVAL
#    define MOUSEKEY_INTERVAL       20
#endif
#ifndef MOUSEKEY_DELAY
#    define MOUSEKEY_DELAY          0
#endif
#ifndef MOUSEKEY_TIME_TO_MAX
#    define MOUSEKEY_TIME_TO_MAX    60
#endif
#ifndef MOUSEKEY_MAX_SPEED
#    define MOUSEKEY_MAX_SPEED      7
#endif
#ifndef MOUSEKEY_WHEEL_DELAY
#    define MOUSEKEY_WHEEL_DELAY    0
#endif
```

This is what I meant, specifically.  This should allow users to set different values, but default to these, if none are set.

File: keyboards/kudox_full/readme.md [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+
+![Kudox Full image2](https://raw.githubusercontent.com/kumaokobo/kudox-keyboard/master/img/kudox-full-image2.jpg) | ![Kudox Full image1](https://raw.githubusercontent.com/kumaokobo/kudox-keyboard/master/img/kudox-full-image1.jpg)
+:----:|:-----:
+ front | angle
+
+- Keyboard Maintainer: [Kumao Kobo](https://github.com/kumaokobo)
+- Hardware Supported: Kudox Full PCB rev1.0 w/ Pro Micro
+- Assembly guide: [Kudox Full/README](https://github.com/kumaokobo/kudox-keyboard/blob/master/kudox-full/README.md)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/kudox_full/readme.md:13]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/kudox_full/rev1/rev1.c [hunk 25]
```diff
@@ -0,0 +1 @@
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/kudox_full/rev1/rev1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kudox_full/post_config.h (hunk 23 above)
Comment (x1-) [reply keyboards/kudox_full/post_config.h:7]: Thank you. That makes sense!

**Comment by x1-** [comment]
@drashna
Thank you for reviewing kindly!
I modified the codes you suggested.

**Review by drashna — APPROVED** [review]

File: keyboards/kudox_full/rules.mk [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/kudox_full/rules.mk:16]:
```suggestion
NKRO_ENABLE = no            Enable N-Key Rollover
```

File: keyboards/kudox_full/rules.mk [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight
```
Comment (noroadsleft) [inline keyboards/kudox_full/rules.mk:18]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by x1-** [comment]
@noroadsleft 
Thank you for reviewing!
Could you re-review, since I modified the codes you suggested.

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**